Lint: 18 ESLint errors break CI on fresh `pnpm install`

## Symptoms

Running `pnpm lint` on a freshly cloned khaopad fails with 18 errors (and 407 warnings). Discovered while rebasing a downstream project (BACtrack Thailand) onto a more recent eslint/eslint-plugin-svelte combo.

## The errors

```
src/lib/components/seo/Seo.svelte
  93:3   error  `{@html}` can lead to XSS attack            svelte/no-at-html-tags
  93:68  error  Unnecessary escape character: \/            no-useless-escape

src/lib/components/consent/CookieBanner.svelte
  49:9   error  Unexpected href link without resolve()      svelte/no-navigation-without-resolve

src/lib/server/webhooks/index.ts
  82:9   error  responseExcerpt assigned but not used in subsequent statements  no-useless-assignment

src/routes/(cms)/cms/dashboard/+page.server.ts
  3:20   error  'and' is defined but never used             @typescript-eslint/no-unused-vars

src/routes/(cms)/cms/media/+page.svelte
  39:15  error  Use SvelteMap instead of plain Map          svelte/prefer-svelte-reactivity

src/routes/(www)/[locale]/[...slug]/+page.svelte
  25:3   error  `{@html}` can lead to XSS attack            svelte/no-at-html-tags

src/routes/(www)/[locale]/+page.svelte
  18:10  error  onMount cleanup return inside async         (related, type-checker)
```

(plus a few related ones in pages I added downstream — those are mine to fix)

## Impact

CI gate (`gate` job in `.github/workflows/deploy.yml` -> `pnpm run check / lint`) blocks every push. New downstream consumers can't run the deploy pipeline at all without first pushing eslint-disable comments.

## Suggested fix

Two options:

1. **Targeted fixes** — drop the unused `and` import, remove the dead `responseExcerpt` assignment, add eslint-disable comments around the legitimate `{@html}` cases (JSON-LD + sanitized markdown), wrap the navigation-without-resolve in `CookieBanner` (it's an external-ish privacy URL).
2. **Loosen the rules globally** — tougher to justify; the rules catch real bugs.

I'd vote (1). Happy to send the PR.

## Related

- Vite plugin `paraglide-js` `strategy: ["url", "cookie", "baseLocale"]` — the cookie tier means a stale visitor cookie can override the URL on bare `/`. Probably worth dropping `cookie` from the default for new installs (or documenting). Found this confusing customers downstream.

--- a/src/lib/components/consent/CookieBanner.svelte
+++ b/src/lib/components/consent/CookieBanner.svelte
@@ -46,6 +46,7 @@
 				<p class="font-medium text-foreground mb-1">{m.cookie_banner_title()}</p>
 				<p>
 					{m.cookie_banner_body()}
+					<!-- eslint-disable-next-line svelte/no-navigation-without-resolve -- privacy href is operator-supplied (always /[locale]/privacy-policy from layout); not a build-time route -->
 					<a href={privacyHref} class="underline hover:text-foreground">{m.cookie_banner_learn_more()}</a>
 				</p>
 				{#if detailsOpen}

--- a/src/lib/components/seo/Seo.svelte
+++ b/src/lib/components/seo/Seo.svelte
@@ -88,9 +88,16 @@
 		<meta name="twitter:site" content={defaults.twitter} />
 	{/if}
 
-	<!-- JSON-LD: one <script> per entry for maximum tooling compatibility -->
+	<!-- JSON-LD: one <script> per entry for maximum tooling compatibility.
+		 Using {@html} is required to emit a literal <script> tag (Svelte
+		 strips them in normal markup). The payload is server-built JSON
+		 from our own typed builders ($lib/seo) — never user input — and
+		 we escape any "</script>" sequence before injection so a JSON
+		 value can't break out of the tag. -->
 	{#each jsonLd as ld, i (i)}
-		{@html `<script type="application/ld+json">${JSON.stringify(ld)}<\/script>`}
+		{@const safeJson = JSON.stringify(ld).replace(/<\/script>/gi, '<\\/script>')}
+		<!-- eslint-disable-next-line svelte/no-at-html-tags -- trusted: server-built JSON-LD with </script> escaped -->
+		{@html '<script type="application/ld+json">' + safeJson + '<' + '/script>'}
 	{/each}
 
 	<!-- RSS auto-discovery -->

--- a/src/lib/server/webhooks/index.ts
+++ b/src/lib/server/webhooks/index.ts
@@ -79,7 +79,9 @@ async function deliverOne(
   for (let attempt = 1; attempt <= MAX_INLINE_ATTEMPTS; attempt++) {
     const t0 = Date.now();
     let responseStatus: number | null = null;
-    let responseExcerpt: string | null = null;
+    // Assigned in every code path below; no initializer to satisfy
+    // no-useless-assignment.
+    let responseExcerpt: string | null;
     let ok = false;
     try {
       const ctrl = new AbortController();

--- a/src/routes/(cms)/cms/dashboard/+page.server.ts
+++ b/src/routes/(cms)/cms/dashboard/+page.server.ts
@@ -1,6 +1,6 @@
 import { redirect, error } from "@sveltejs/kit";
 import { drizzle } from "drizzle-orm/d1";
-import { desc, eq, and, gte } from "drizzle-orm";
+import { desc, eq, gte } from "drizzle-orm";
 import * as schema from "$lib/server/content/schema";
 import { canManageUsers } from "$lib/server/auth/permissions";
 import { AnalyticsService } from "$lib/server/analytics";

--- a/src/routes/(cms)/cms/media/+page.svelte
+++ b/src/routes/(cms)/cms/media/+page.svelte
@@ -34,8 +34,11 @@
 		Boolean(data.user && ['super_admin', 'admin', 'editor'].includes(data.user.role)),
 	);
 
-	// Build a parent → children map for the folder tree.
+	// Build a parent → children map for the folder tree. Local to the
+	// derived; never read reactively (rebuilt every invocation), so a
+	// plain Map is correct — no need for SvelteMap's reactivity overhead.
 	const childrenByParent = $derived.by(() => {
+		// eslint-disable-next-line svelte/prefer-svelte-reactivity -- local lookup, not reactive state
 		const map = new Map<string | null, MediaFolderRecord[]>();
 		for (const f of data.folders) {
 			const parent = f.parentId ?? null;

--- a/src/routes/(www)/[locale]/[...slug]/+page.svelte
+++ b/src/routes/(www)/[locale]/[...slug]/+page.svelte
@@ -22,6 +22,7 @@
 		<h1 class="text-4xl font-bold mb-2">{data.title}</h1>
 	</header>
 	<div class="prose prose-neutral dark:prose-invert max-w-none">
+		<!-- eslint-disable-next-line svelte/no-at-html-tags -- trusted: server-rendered markdown from CMS -->
 		{@html data.htmlContent}
 	</div>
 </article>

